rust/vectordb/src/data/inspect.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use arrow::compute::kernels::{aggregate::bool_and, length::length};
+use arrow_array::{cast::AsArray, types::Int32Type, Array, RecordBatchReader};
+use arrow_ord::comparison::eq_dyn_scalar;
+use arrow_schema::DataType;
+
+use crate::error::{Error, Result};
+
+pub(crate) fn infer_dimension(arr: &dyn Array) -> Result<Option<i32>> {
```

Comment:
should we make this a generic function so we can customize the data width?

Reply:
I can just make it work with both list/large list? the data width of each element should be not relevant i think.

Reply:
https://docs.rs/arrow-array/46.0.0/arrow_array/array/struct.GenericListArray.html